src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java

```diff
@@ -395,4 +401,14 @@ public DetailAST getFirstChild() {
         return (DetailAST) super.getFirstChild();
     }
 
+    /**
+     * Clears lazy-load cache for all parents of the current {@link DetailAST} instance.
+     */
+    private void clearLazyLoadCache() {
+        DetailAST prevParent = getParent();
+        while (prevParent != null) {
+            prevParent.branchTokenTypes = null;
```

Comment:
`clearLazyLoadCache`  vs `branchTokenTypes = null;`

it is not the only cache that we have in that class , so please be specific in the name - `clearBranchTokenTypesCache`